src/raster/warp.rs

```diff
@@ -25,3 +29,36 @@ pub fn reproject(src: &Dataset, dst: &Dataset) -> Result<()> {
     }
     Ok(())
 }
+
+pub fn create_and_reproject(
+    src: &Dataset,
+    dst_path: &Path,
+    dst_srs: &SpatialRef,
+    dst_driver: &Driver,
+) -> Result<()> {
+    let psz_dst_filename = dst_path
+        .to_str()
+        .expect("Destination path must be supplied.");
+    let psz_dst_wkt = dst_srs.to_wkt().expect("Failed to obtain WKT for SRS.");
+
+    let rv = unsafe {
+        gdal_sys::GDALCreateAndReprojectImage(
+            src.c_dataset(),
+            null(),
+            CString::new(psz_dst_filename)?.as_ptr(),
```

Comment:
This is a use-after-free: the `CString` is destroyed before `GDALCreateAndReprojectImage` runs.